### PR TITLE
Fixed critical Sonar issues in config packages

### PR DESCRIPTION
**Pull request by vbergmann**

### CODE REVIEW

File: src/main/java/org/aludratest/config/impl/SimplePreferences.java [hunk 1]
```diff
@@ -29,11 +29,16 @@
  */
 public class SimplePreferences extends AbstractPreferences implements MutablePreferences {
 
-    private Map<String, String> values = new HashMap<String, String>();
+    private Map<String, String> values;
```
Comment (falbrech-hsdg) [inline src/main/java/org/aludratest/config/impl/SimplePreferences.java]: Why this change? Is this a SonarQube recommendation? The compiler will generate exactly that constructor you added manually...

Comment (vbergmann) [inline src/main/java/org/aludratest/config/impl/SimplePreferences.java]: Sonar's rule squid:S1258 wants a constructor that initializes the class' attributes.

Comment (falbrech-hsdg) [inline src/main/java/org/aludratest/config/impl/SimplePreferences.java]: Looks like a False Positive: Non-abstract classes and enums with non-static, private members should explicitly initialize those members, either in a constructor **or with a default value**.

Comment (falbrech-hsdg) [inline src/main/java/org/aludratest/config/impl/SimplePreferences.java]: No, have to correct myself. According to the description, I bet this relates to children, not to values.

Comment (vbergmann) [inline src/main/java/org/aludratest/config/impl/SimplePreferences.java]: Anyway I consider this good coding style to initialize all non-static attributes only in the constructor (which is BTW my default approach): You have all initializations in a single place, do not get messed up when changing the order of the attributes and other users are less likely to misunderstand trickier parts of the initialization. OK, for a single attribute this doesn't matter a lot, but I would adhere to the rule anyway.
